modules/core/src/main/scala/org/tesselation/Main.scala

```diff
@@ -27,31 +32,39 @@ object Main extends IOApp {
     Config.load[IO].flatMap { cfg =>
       logger.info(s"Config loaded") >>
         logger.info(s"App environment: ${cfg.environment}") >>
-        SecurityProvider.forAsync[IO].use { implicit securityProvider =>
-          loadKeyPair(cfg.keyConfig).flatMap { keyPair =>
-            KryoSerializer.forAsync[IO](coreKryoRegistrar).use { implicit kryoPool =>
-              DoobieTransactor.forAsync[IO](cfg.dbConfig).use { implicit doobieTransactor =>
-                Supervisor[IO].use { _ =>
-                  (for {
-                    res <- AppResources.make[IO](cfg)
-                    nodeId = PeerId.fromPublic(keyPair.getPublic)
-                    p2pClient = P2PClient.make[IO](res.client)
-                    storages <- Resource.eval {
-                      Storages.make[IO]
-                    }
-                    services <- Resource.eval {
-                      Services.make[IO](cfg, nodeId, keyPair, storages)
-                    }
-                    programs <- Resource.eval { Programs.make[IO](storages, services, p2pClient, nodeId) }
-                    api = HttpApi.make[IO](storages, services, programs, cfg.environment)
-                    _ <- MkHttpServer[IO].newEmber(ServerName("public"), cfg.httpConfig.publicHttp, api.publicApp)
-                    _ <- MkHttpServer[IO].newEmber(ServerName("p2p"), cfg.httpConfig.p2pHttp, api.p2pApp)
-                    _ <- MkHttpServer[IO].newEmber(ServerName("cli"), cfg.httpConfig.cliHttp, api.cliApp)
-                  } yield ()).useForever
+        parser.parse[IO](args).flatMap { cli =>
+          SecurityProvider.forAsync[IO].use { implicit securityProvider =>
+            loadKeyPair(cfg.keyConfig).flatMap { keyPair =>
+              KryoSerializer.forAsync[IO](coreKryoRegistrar).use { implicit kryoPool =>
+                DoobieTransactor.forAsync[IO](cfg.dbConfig).use { implicit doobieTransactor =>
+                  Supervisor[IO].use { _ =>
+                    (for {
+                      res <- AppResources.make[IO](cfg)
+                      nodeId = PeerId.fromPublic(keyPair.getPublic)
+                      p2pClient = P2PClient.make[IO](res.client)
+                      storages <- Resource.eval(Storages.make[IO])
+                      services <- Resource.eval(Services.make[IO](cfg, nodeId, keyPair, storages))
+                      programs <- Resource.eval(Programs.make[IO](storages, services, p2pClient, nodeId))
+                      _ <- Resource.eval {
+                        cli.method match {
+                          case CliMethod.RunValidator => IO.unit
+                          case CliMethod.RunGenesis =>
+                            GenesisLoader.make[IO].load(Path(cli.genesisPath)).flatMap { accounts =>
```

Comment:
What is the type of `genesisPath`? Maybe you could cast it to `Path` because it probably won't be used as regular string. The same we do with `Url` itd.

Reply:
:+1: